// futurapay/futurapay.js

const Encryptions = require('./Encryptions'); // Adjust the path as necessary
const PaymentURL = require("./PaymentURL")

class Futurapay {

    envirment = "sandbox"
    paymentType = "deposit"
    constructor(merchantKey, apiKey, siteId) {
        this.merchantKey = merchantKey
        this.apiKey = apiKey
        this.siteId = siteId
    }

    setEnv = (envirment) => {
        this.envirment = envirment
    }
    setType = (paymentType) => {
        this.paymentType = paymentType
    }

    /*  @payload  must be type of array  */
    initiatePayment = (payload) => {
        payload['merchant_key'] = this.merchantKey
        payload['site_id'] = this.siteId
        payload['api_key'] = this.apiKey

        const result = Encryptions.make(this.merchantKey, this.apiKey, this.siteId, payload);
        const urlString = new URLSearchParams(result).toString();
        const paymentURl = this.getPaymentUrl();
        const final_encrypted_url = paymentURl + urlString
        return final_encrypted_url

    }

    getPaymentUrl = () => {
        let url = ''
        if (this.envirment == "live") {
            url = PaymentURL.LIVE_DEPOSIT_URL
            if (this.paymentType == "withdraw") {
                url = PaymentURL.LIVE_WITHDRAWAL_URL
            }
        } else {
            url = PaymentURL.STAGE_DEPOSIT_URL
            if (this.paymentType == "withdraw") {
                url = PaymentURL.STAGE_WITHDRAWAL_URL
            }
        }
        return url
    }

}

module.exports = Futurapay
